### ISSUE TITLE
When bidding, bot uses the wrong pricing method.

### ISSUE THREAD

**Issue by steenburgh** [azerothcore/mod-ah-bot#28]
Perhaps I'm mistaken, but it looks like this line of code was meant to have a `!`. I'd make a PR but I'm not familiar enough with the code to be sure I'm correct.

The config says: 
```
#    AuctionHouseBot.UseBuyPriceForBuyer
#        Should the Buyer use BuyPrice or SellPrice to determine Bid Prices
#    Default 0 (use SellPrice)
```

But the code seems to do the opposite:
https://github.com/azerothcore/mod-ah-bot/blob/45cdc1cd8a5862e0088de1c3ef1fb990493c2768/src/AuctionHouseBot.cpp#L569
```
if (BuyMethod) 
{
    ...
    prototype->SellPrice
}
else 
{
    ...
    prototype->BuyPrice
}

**Comment by mserajnik** [comment]
This seems to still be an issue:
https://github.com/azerothcore/mod-ah-bot/blob/c34c10a935506ea998485e10eec18d9efef14086/src/AuctionHouseBot.cpp#L566-L596
It is checked if `BuyMethod` is true, which is the case when the configuration `AuctionHouseBot.UseBuyPriceForBuyer` is set to true.

But then it proceeds to use `prototype->SellPrice` there and `prototype->BuyPrice` when `BuyMethod` is false, so this seems to be reversed by mistake.

Can one of the maintainers confirm if this is correct or not? @Winfidonarleyan perhaps?

**Comment by Winfidonarleyan** [comment]
Remind me to watch on Saturday, Discord or here via ping me

**Comment by WuStangDan** [comment]
Yes this bug is still here. I tested it in game to confirm (after being suspicious that a few high sells were going through) and yes even when setting in the config `AuctionHouseBot.UseBuyPriceForBuyer = 0` it uses the buy price, not the sell price for the buyer. 

This means that there are plenty of situations where you could create a money pump with this bot. Buying items from the bot and selling back at a much higher price.

Love the mod overall though, thanks.
